### PR TITLE
"Naive string search" in C#

### PR DESCRIPTION
**Pull request by vhoudoverdov**
Greetings,

I was interested in contributing a simple algorithm for the Algorithm.Strings component of this repo.  The algorithm is the naive string search algorithm, cost O(n*m).   The naive string search can also be found in a few other language repositories across the TheAlgorithms project.

Regards,
Vasken

### CODE REVIEW

File: Algorithms/Strings/NaiveStringSearch.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+using System;
+
+namespace Algorithms.Strings
+{
+    /// <summary>
+    /// Implements the traditional naive string matching algorithm in C#.
+    /// Cost:  O(n*m), where m is pattern length and n is content length.
+    /// </summary>
+    class NaiveStringSearch
```
Comment (siriak) [inline Algorithms/Strings/NaiveStringSearch.cs]: Codacy Issue found: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

File: Algorithms/Strings/NaiveStringSearch.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            int j;
+
+            for (j = 0; j < m; j++)
+                {
+                  if (content[e + j] != pattern[j])
+                  break;
+                 }
+
+            if (j == m)
+                Console.WriteLine("Pattern occurs in content at index " + e);
```
Comment (siriak) [inline Algorithms/Strings/NaiveStringSearch.cs]: Codacy Issue found: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

File: Algorithms/Strings/NaiveStringSearch.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
```
Comment (siriak) [inline Algorithms/Strings/NaiveStringSearch.cs]: Codacy Issue found: [Remove this logging statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by vhoudoverdov** [comment]
Introduced static keyword to class declaration.
Replaced console logging with list-based return of all relevant indices.

**Comment by vhoudoverdov** [comment]
Introduced static keyword to class declaration.
Replaced console logging with list-based return of all relevant indices.

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

File: Algorithms/Strings/NaiveStringSearch.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+
+/// Implements the traditional naive string matching algorithm in C# for TheAlgorithms/C-Sharp.
+namespace Algorithms.Strings
+{
+   public static class NaiveStringSearch
+   {
+	 		/// <summary>
```
Comment (siriak) [inline Algorithms/Strings/NaiveStringSearch.cs]: Codacy Issue found: [Replace all tab characters in this file by sequences of white-spaces.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by vhoudoverdov** [comment]
Hi @siriak.  Sorry for the delay in fixing the issues and stylecop findings.  I had to familiarize myself with some additional C# documentation practices.  I've got the code passing the Travis CI tests now.  Codecov is complaining now even though it was previously passing.

**Comment by vhoudoverdov** [comment]
Hey, @siriak.  I'll work on contributing some tests, as it will give me a chance to experiment a bit more.  I'll reference existing tests in the repo.  Thank you!

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Algorithms.Tests/Strings/NaiveStringSearchTests.cs  1
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

File: Algorithms.Tests/Strings/NaiveStringSearchTests.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+   {
+      [Test]
+      public static void ThreeMatchesFound_PassExpected()
+      {
+         // Arrange
+         var pattern = "ABB";
+         var content = "ABBBAAABBAABBBBAB";
+
+         // Act
+         var expectedOccurrences = new int[] {0, 6, 10};
```
Comment (siriak) [inline Algorithms.Tests/Strings/NaiveStringSearchTests.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632)

File: Algorithms.Tests/Strings/NaiveStringSearchTests.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+      [Test]
+      public static void OneMatchFound_PassExpected()
+      {
+         // Arrange
+         var pattern = "BAAB";
+         var content = "ABBBAAABBAABBBBAB";
+
+         // Act
+         var expectedOccurrences = new int[] {8};
```
Comment (siriak) [inline Algorithms.Tests/Strings/NaiveStringSearchTests.cs]: Codacy Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms.Tests/Strings/NaiveStringSearchTests.cs  1
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms.Tests/Strings/NaiveStringSearchTests.cs  1
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Comment by vhoudoverdov** [comment]
Hi @siriak.  I contributed a few unit tests for the algorithm and obtained full coverage with Codecov .  I am trying to learn more about why Codecov thinks that FastSearcher.cs is an impacted file in this scenario (I'm still new to Codecov) because that file is unrelated to this PR.  I'm guessing it has to do with the execution of the unit testing process.

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms.Tests/Strings/NaiveStringSearchTests.cs  1
- Algorithms/Strings/NaiveStringSearch.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4766632&bid=15888839)

**Pull request merged**